### DISCUSSION TITLE
Plugin Validation issues with maven-resources:3.3.0 and maven-compiler-plugin:3.9.0 or 3.11.0

### DISCUSSION THREAD

**Discussion by independentid** [quarkusio/quarkus#33960] [Q&A]
Recently for a previously compiling project I started getting validation issues related to quarkus and maven plugins.

When I run  mvn compile -Dmaven.plugin.validation=VERBOSE

I get the following errors:
[WARNING] 
[WARNING] Plugin validation issues were detected in 2 plugin(s)
[WARNING] 
[WARNING]  * org.apache.maven.plugins:maven-compiler-plugin:3.9.0
[WARNING]   Declared at location(s):
[WARNING]    * com.independentid:i2scim-parent:0.7.0-Alpha (pom.xml) @ line 100
[WARNING]   Used in module(s):
[WARNING]    * com.independentid:i2scim-core:0.7.0-Alpha (i2scim-core/pom.xml)

<snip>

[WARNING]   Plugin issue(s):
[WARNING]    * Plugin should declare these Maven artifacts in `provided` scope: [org.apache.maven:maven-plugin-api:3.2.5, org.apache.maven:maven-model:3.2.5, org.apache.maven:maven-core:3.2.5, org.apache.maven:maven-settings:3.2.5, org.apache.maven:maven-model-builder:3.2.5, org.apache.maven:maven-repository-metadata:3.2.5, org.apache.maven:maven-aether-provider:3.2.5, org.apache.maven:maven-settings-builder:3.2.5, org.apache.maven:maven-artifact:3.2.5]
[WARNING] 
[WARNING]  * org.apache.maven.plugins:maven-resources-plugin:3.3.0
[WARNING]   Declared at location(s):
[WARNING]    * org.apache.maven:maven-core:3.9.2:default-lifecycle-bindings @ line -1
[WARNING]   Used in module(s):
[WARNING]    * com.independentid:i2scim-core:0.7.0-Alpha (i2scim-core/pom.xml)

<snip>

[WARNING]   Mojo issue(s):
[WARNING]    * Mojo resources:resources (org.apache.maven.plugins.resources.ResourcesMojo)
[WARNING]      - Implements `Contextualizable` interface from Plexus Container, which is EOL.
[WARNING] 
[WARNING] 
[WARNING] Fix reported issues by adjusting plugin configuration or by upgrading above listed plugins. If no upgrade available, please notify plugin maintainers about reported issues.
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING] 

Note: I recently upgraded mvn to the most recent (3.9.2) using brew upgrade.  I have tried different maven builder settings in IntelliJ

The error that comes up most consistently are related to Mojo ??

I am trying to run the latest Quarkus 3.1.1.Final platform, but it is not clear what versions the jandex and maven-compiler  build plugins should actually be.

[a comment by an automated account is not shown]

**Comment by independentid** [comment]
I have traced the issue. It appears maven 3.9.2 is not compatible.  If brew upgrade is used, mac users get forced into 3.9.2.  Brew does not have an installer for 3.8.8 any more.

I noted the mvnw executable that comes with Start Coding download is version 3.8.8.  When I manually installed Maven 3.8.8.

Assuming I don't have some other problem, this would be well worth noting as an issue for now.

**Comment by aloubyansky** [comment]
The quarkus-maven-plugin isn't on the list though. As the message suggests, these issues should be reported to the maintainers of the listed plugins. Although, by now, I suppose they are already aware of them.
Note, these are not errors but warnings and you can keep using Maven 3.9.2 for your project. It just became a lot more verbose reporting these kinds of issues that were there before for years.
